addons/account/tests/test_account_analytic.py

```diff
@@ -264,3 +264,29 @@ def test_cross_analytics_computing(self):
         self.assertEqual(self.analytic_account_a.invoice_count, 1)
         self.analytic_account_a._compute_vendor_bill_count()
         self.assertEqual(self.analytic_account_a.vendor_bill_count, 1)
+
+    def test_applicability_score(self):
+        """ Tests which applicability is chosen if several ones are valid """
+        applicability_without_company, applicability_with_company = self.env['account.analytic.applicability'].create([
+            {
+                'business_domain': 'invoice',
+                'product_categ_id': self.product_a.categ_id.id,
+                'applicability': 'mandatory',
+                'analytic_plan_id': self.default_plan.id,
+                'company_id': False,
+            },
+            {
+                'business_domain': 'invoice',
+                'applicability': 'unavailable',
+                'analytic_plan_id': self.default_plan.id,
+                'company_id': self.env.company.id,
+            },
+        ])
+
+        applicability = self.default_plan._get_applicability(business_domain='invoice', company_id=self.env.company.id, product=self.product_a.id)
+        self.assertEqual(applicability, 'mandatory', "product takes precedence over company")
+
+        score = applicability_without_company._get_score(business_domain='invoice', product=self.product_a.id)
+        self.assertEqual(score, 2, "company_id shouldn't be taken into account")
+        score = applicability_with_company._get_score(business_domain='invoice', product=self.product_a.id)
+        self.assertEqual(score, 1, "company_id shouldn't be taken into account either")
```

Comment:
I find the message a bit misleading here (especially the "either"). Not sure those two (this one and the one above) make the test any more explicit. Wouldn't it be just better, to leave those two assertEqual without error message ? The "product takes precedence over company" above already states the expected behavior, if I get the test well :thinking:

Reply:
Not exactly, we're not really testing the same thing.
But still, I reworded the message to make clearer what I wanted to test in these 2 asserts as indeed, it wasn't explicit